features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -93,17 +95,51 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
-        val base64String = convertBmpToBase64(drawable, bitmap, shouldCacheBitmap)
-        finalizeRecordedDataItem(base64String, imageWireframe, base64SerializerCallback)
+        val byteArray = compressBitmap(bitmap)
+        val base64String = convertBmpToBase64(byteArray, drawable, bitmap, shouldCacheBitmap)
+        val hash = calculateHash(drawable, byteArray)
+        finalizeRecordedDataItem(base64String, hash, imageWireframe, base64SerializerCallback)
+    }
+
+    private fun calculateHash(
+        drawable: Drawable,
+        byteArray: ByteArray
+    ): String {
+        val md5Generator = MD5HashGenerator(logger)
+        val hashValue = md5Generator.generate(byteArray)
+
+        if (hashValue.isNotEmpty()) {
+            hashLRUCache?.put(drawable, hashValue)
+        }
+
+        return hashValue
     }
 
     @MainThread
-    private fun registerCacheForCallbacks(applicationContext: Context) {
-        if (isCacheRegisteredForCallbacks) return
+    private fun registerBase64LruCacheForCallbacks(applicationContext: Context) {
+        if (isBase64CacheRegisteredForCallbacks) return
```

Comment:
why do we still keep/use the base64LruCache ? Can't we just keep a cache of the binary instead with a key which is the `resourceId` ?

Reply:
even if it is needed, would it make sense to use a single cache for both base64 and hash? or it will make things more complicated?

Reply:
@mariusc83 we still send base64 and will only stop sending it when the rss endpoint is up, so I think we still need to cache the values to prevent unnecessary computations. 

@0xnm You mean a single cache that maps a drawable to a pair of base64 and md5? The issue that I see is that it couples the two hashes and forces us to change the cache implementation now, and then change it again when we want to remove b64 hash.

Reply:
I would rather wait on this PR and remove base64 all together when we will be able to send binaries. What blocks you ?

Reply:
RSS endpoint is still a wip on the backend. On iOS they are already sending resourceId alongside b64 as preparation for the next stage, and this pr is mainly to align on that

Reply:
are we sure we are not going to do this PR for nothing now just to align with iOS ? In the end once the endpoint will be in place we will have to get back and remove this and re - factor. Why should we not wait ? How much of this PR work is going to help you when the endpoint will be ready ?

Reply:
There's a benefit in already removing base64 from MutationResolver. We expect to have the same level of benefit in terms of diffing speed that iOS achieve. Remember, this GitHub [issue](https://github.com/DataDog/dd-sdk-ios/issues/1482) originally triggered the investigation. 

Yes we'll have to revisit it in the future, but we still don't know when. Likely only next year and on the mean time it's beneficial to have some performance improvements and alignment with iOS.